workspace/__ardulib__/XColorPattern/XColorPattern.h

```diff
@@ -0,0 +1,46 @@
+#ifndef __X_COLOR_PATTERN_H__
+#define __X_COLOR_PATTERN_H__
```

Comment:
Wouldn’t it be much simpler to embed the types to the constructor node implementation instead of a dedicated “Third party” library?

Reply:
Until the `WS2812` knows something about `Pattern` type (even if I implement `fill-pattern` node inside NIIX instead of calling `fillPattern` method) — NO.

However, if I combine XColorPattern with the `WS2812` (like you said in [https://github.com/xodio/xod/pull/1937#discussion_r366311550](previous comment)) I can combine the library too, so `XColorPattern.h` will be a part of `WS2812_NoBuffer` c++ library.

Reply:
> so XColorPattern.h will be a part of WS2812_NoBuffer c++ library.

Sounds good